### DISCUSSION TITLE
Port zephyr on bao

### DISCUSSION THREAD

**Discussion by wjhtinger** [bao-project/bao-hypervisor#33] [General]
I want to run zephyr(RTOS) on bao instead of freertos. So, I want to know if I need to do some modify on zephyr code? For simple, firtsly dont consider the divece driver.

When you run freertos on bao, do you do some porting work on freertos?

https://github.com/zephyrproject-rtos/zephyr

**Comment by josecm** [comment]
> So, I want to know if I need to do some modify on zephyr code?

I would say no. I'm not that familiar with Zephyr, but I believe it uses a device tree description at build time to setup the build for a given platform. In this sense, I guess you'd need to provide it with a trimmed device tree with only those resources you assign it in the bao configuration, the same you would do for a linux guest.

> When you run freertos on bao, do you do some porting work on freertos?

We did have to "port" freertos to bao because the original freertos ports target a specific platform. We modify it to target a generic armv8-a platform. However, I wouldn't think you'd need to do this for zephyr.
